api/src/main/java/jakarta/data/metamodel/BasicAttribute.java

```diff
@@ -52,8 +51,7 @@ default Restriction<T> notIn(Set<V> values) {
     }
 
     default Restriction<T> notNull() {
-        //TODO: fix this "<> NULL" is not the same thing as "IS NOT NULL" in JDQL
-        return Restrict.notEqualTo(null, name());
+        return Restrict.isNotNull(name());
```

Comment:
Recommend switching `isNotNull` to `notNull` in order to align with our other names: `notEqualTo`, `notIn`, `notNull` (from BasicAttribute), `notContains`, `notEndsWith`, `notLike`, `notStartsWith`.  

```suggestion
        return Restrict.notNull(name());
```

Reply:
So I'm not sure what to do here. The issue is that since `null` is a keyword and can't be a method name, it has to be `isNull`. But then `isNull`/`notNull` doesn't conform with the pattern.

Reply:
> So I'm not sure what to do here. The issue is that since `null` is a keyword and can't be a method name, it has to be `isNull`. But then `isNull`/`notNull` doesn't conform with the pattern.

Yeah, I'm not sure either.  At least for now, I'd say name them `Restrict.isNull` and `Restrict.notNull` to be consistent with the `BasicAttribute.isNull` and `BasicAttribute.notNull` methods that we already have.  Then our spec will at least match in case we forget to revisit the naming. And if we do decide to change them later, we can change them all together.

Reply:
OK, fine. I can't think of anything better.

Reply:
Done.